### DISCUSSION TITLE
This is amazing, thank you, and amazing work!

### DISCUSSION THREAD

**Discussion by Yohn** [givanz/Vvveb#139] [General]
All I can say is wow. This is amazing, and I've been looking through the source last night and today and really like how you built this system. I've been contemplating on bring my old script back to life that I wrote about 15 years ago and I'm sure you know that process wouldn't be pretty.

So I'm looking at what you got, and I'm interested in possibly helping get off the ground a little more. I know there's a lot that does still need to be done. I also know what you provided here for download is not what you have completed on the demo site since the demo site does have a few more features - completely understandable considering the scope of  this project.

Anyways, I'm wondering what features you might like someone else to try to tackle? I'll be happy to take a look more into getting some of these features working instead of adding new ones, or working on something you already have almost ready to be released. 

One  thing that I would like to see sooner, rather than later, is the Media Library so it'll be easier to post pictures with everything. Another thing I saw, when all the available plugins are activated, the pages and posts are showing the HTML code instead of the actual format that was expected.

**Comment by givanz** [comment]
Thank you.

> I know there's a lot that does still need to be done

Most essential features are done, there are still some bug reports mostly caused by different system/php/db configurations that are fixed.

Automatic update will soon be enabled for testing updates.

Currently I'm working on the SEO plugin.

A rough roadmap with the folllowing features that will be implemented: 

* Advanced seo features like schema markup for seo plugin
* Marketplace expansion and improvements
* Better layout options for page builder
* REST Api
* Custom fields
* More ecommerce features like advanced filtering
* More components and improvements for the page builder controls
* Multi vendor plugin.
* Advanced features for page builder like animation controls, better controls for shadows, margins, padding, flex, grid etc

---

> the demo site does have a few more features

The demo is the same as latest.zip, it's periodically reinstalled from the zip for testing.

The only difference is that the demo has some plugins such as payment methods that are not included in the zip or available in the market because they are in a very early stage and not usable yet, they are installed in the demo for testing.

When finished they will be added to the plugins marketplace and available for install from `Add new plugin`.

There are also plugins that are not bundled but available for install from the marketplace.

----

> One thing that I would like to see sooner, rather than later, is the Media Library so it'll be easier to post pictures with everything

What features would make posting pictures better in media library? Do you have problems with file uploads in media library?

What is planned next for media library is editing image alt description that is useful for seo, insert image gallery option for text editor (tinymce), media type filtering to select only videos/images etc, better video support and media preview and better managemenet like folder creation.

There is also a plan for a plugin for storing and serving all media in the cloud such as S3 and also plugins for searching and inserting stock images from sites such as unsplash, pixabay etc directly from the media library.

---

> when all the available plugins are activated, the pages and posts are showing the HTML code instead of the actual format that was expected.

I think that this is caused by the headless plugin which returns json instead of the regular page, this plugin is only provided for testing headless themes until REST Api is finished, you can disable it to show regular pages.

**Reply by Yohn** [reply]
> Most essential features are done, there are still some bug reports mostly caused by different system/php/db configurations that are fixed.

I currently just tested locally with Laragon, and just had to enable the gettext extension and then reinstalled via the [GitHub CLI](https://github.com/givanz/Vvveb) and then followed your steps to build locally. 
> There are also plugins that are not bundled but available for install from the marketplace.

Yeah, I installed and activated every plugin that was available
> > One thing that I would like to see sooner, rather than later, is the Media Library so it'll be easier to post pictures with everything
> 
> What features would make posting pictures better in media library? Do you have problems with file uploads in media library?

I was unable to upload any pictures, I would click the edit photo on the products, or posts nothing happens, and go to the media library and tap the upload button but nothing happens after I select my photos. It just sits there with nothing uploading.
My current Laragon setup is using PHP 8.1

> > when all the available plugins are activated, the pages and posts are showing the HTML code instead of the actual format that was expected.
> 
> I think that this is caused by the headless plugin which returns json instead of the regular page, this plugin is only provided for testing headless themes until REST Api is finished, you can disable it to show regular pages.

I think it might have been from one of the markdown editor plugins, but I'm not completely sure

**Reply by Yohn** [reply]
> What features would make posting pictures better in media library? Do you have problems with file uploads in media library?

Looking at the console from within the Pages -> edit page, I'm seeing
`Uncaught ReferenceError: MediaModal is not defined`
when I click on the current picture that is there, and when I click "Set features media"

**Reply by givanz** [reply]
> Looking at the console from within the Pages -> edit page, I'm seeing
> Uncaught ReferenceError: MediaModal is not defined
> when I click on the current picture that is there, and when I click "Set features media"

Thanks for feedback, I tried on a fresh install both using latest.zip and 

```bash
git clone --recurse-submodules https://github.com/givanz/Vvveb vvveb
```

but I'm unable to reproduce the MediaModal bug and file upload issue.

I will try to install Laragon on a Windows virtual machine and test maybe this is caused by some Laragon configuration.
